src/EditorFeatures/Core/Shared/Options/FeatureOnOffOptions.cs

```diff
@@ -6,46 +6,58 @@ namespace Microsoft.CodeAnalysis.Editor.Shared.Options
 {
     internal static class FeatureOnOffOptions
     {
-        public const string OptionName = "EditorFeaturesOnOff";
-
         [ExportOption]
-        public static readonly PerLanguageOption<bool> EndConstruct = new PerLanguageOption<bool>(OptionName, "End Construct", defaultValue: true);
+        public static readonly PerLanguageOption<bool> EndConstruct = new PerLanguageOption<bool>(nameof(FeatureOnOffOptions), "End Construct", defaultValue: true,
+            persistences: new RoamingProfilePersistence("TextEditor.%LANGUAGE%.Specific.AutoEndInsert"));
 
+        // This value is only used by Visual Basic, and so is using the old serialization name that was used by VB.
         [ExportOption]
-        public static readonly PerLanguageOption<bool> AutomaticInsertionOfAbstractOrInterfaceMembers = new PerLanguageOption<bool>(OptionName, "Automatic Insertion of Abstract or Interface Members", defaultValue: true);
+        public static readonly PerLanguageOption<bool> AutomaticInsertionOfAbstractOrInterfaceMembers = new PerLanguageOption<bool>(nameof(FeatureOnOffOptions), "Automatic Insertion of Abstract or Interface Members", defaultValue: true,
+            persistences: new RoamingProfilePersistence("TextEditor.%LANGUAGE%.Specific.AutoRequiredMemberInsert"));
 
         [ExportOption]
-        public static readonly PerLanguageOption<bool> LineSeparator = new PerLanguageOption<bool>(OptionName, "Line Separator", defaultValue: false);
+        public static readonly PerLanguageOption<bool> LineSeparator = new PerLanguageOption<bool>(nameof(FeatureOnOffOptions), "Line Separator", defaultValue: false,
+            persistences: new RoamingProfilePersistence(language => language == LanguageNames.VisualBasic ? "TextEditor.%LANGUAGE%.Specific.DisplayLineSeparators" : "TextEditor.%LANGUAGE%.Specific.Line Separator"));
```

Comment:
basically, with this approach, all non-VB languages will get C# behavior.  I think that's ok, but i just want to make sure that's ok with you.